Source/WebCore/platform/network/mac/ResourceErrorMac.mm

```diff
@@ -159,14 +168,16 @@ - (void)encodeWithCoder:(NSCoder *)coder
     m_domain = [m_platformError domain];
     m_errorCode = [m_platformError code];
 
-    if (NSString* failingURLString = [[m_platformError userInfo] valueForKey:@"NSErrorFailingURLStringKey"])
+    RetainPtr userInfo = [m_platformError userInfo];
+    if (auto *failingURLString = dynamic_objc_cast<NSString>([userInfo valueForKey:@"NSErrorFailingURLStringKey"]))
         m_failingURL = URL { failingURLString };
-    else
-        m_failingURL = URL((NSURL *)[[m_platformError userInfo] valueForKey:@"NSErrorFailingURLKey"]);
+    else if (auto *failingURL = dynamic_objc_cast<NSURL>([userInfo valueForKey:@"NSErrorFailingURLKey"]))
```

Comment:
NIT: Should we have an `else m_failingURL = { }` in case that there also isn't a `NSErrorFailingURLKey`?  i.e. moving this inside an `if` could have slightly different behavior than before

Reply:
Oh — while it _is_ a behavior change, I opted for this approach since the old behavior (directly casting the `NSErrorFailingURLKey` entry to an `NSURL`) would just result in ObjC type confusion anyways 😬.